src/hypothesis/strategies.py

```diff
@@ -1064,18 +1064,16 @@ def decimals(min_value=None, max_value=None,
     # Set up a strategy for finite decimals
     if places is not None:
         # Fixed-point decimals are basically integers with a scale factor
-
-        def try_quantize(d):
-            try:
-                return d.quantize(factor)
-            except InvalidOperation:  # pragma: no cover
-                return None
+        # plus complicated issues about context and precision
         factor = Decimal(10) ** -places
-        max_num = max_value / factor if max_value is not None else None
-        min_num = min_value / factor if min_value is not None else None
-        strat = integers(min_value=min_num, max_value=max_num)\
-            .map(lambda d: try_quantize(d * factor))\
-            .filter(lambda d: d is not None)
+        default = Decimal(10) ** (getcontext().prec - places) - factor
```

Comment:
Hmm.  I'll have to think carefully about this, but my first instinct is to accept an explicit `context=` argument defaulting to Python's default context.  Or possibly move to a class that can handle this properly in the do_draw method.  More on this later.